File: packages/sdk/kotlin/core/src/commonMain/kotlin/dev/edgebase/sdk/core/ContextManager.kt
// EdgeBase Kotlin SDK — Context manager.
//
// Manages multi-tenancy context (isolateBy).
//: auth.id is silently filtered — server extracts from JWT only.

package dev.edgebase.sdk.core

import kotlinx.coroutines.sync.Mutex
import kotlinx.coroutines.sync.withLock

/**
 * Context manager for multi-tenancy support.
 *
 * The `auth.id` key is silently filtered from context  * the server extracts it from JWT only.
 *
 * Usage:
 * ```kotlin
 * client.setContext(mapOf("tenantId" to "tenant1"))
 * ```
 */
class ContextManager {
    private val mutex = Mutex()
    private var context: Map<String, Any> = emptyMap()

    // Filter out auth.id — server extracts from JWT only
    suspend fun setContext(ctx: Map<String, Any>) = mutex.withLock {
        context = ctx.filterKeys { it != "auth.id" }
    }
    suspend fun getContext(): Map<String, Any> = mutex.withLock { context }
    suspend fun clearContext() = mutex.withLock { context = emptyMap() }
}

File: packages/sdk/python/src/edgebase/context_manager.py
"""EdgeBase Python SDK — legacy isolateBy context compatibility."""

from __future__ import annotations

from typing import Any


class ContextManager:
    """Stores legacy isolateBy context without serializing HTTP headers.

    The 'auth.id' key is silently filtered — it is set server-side from the JWT.
    """

    def __init__(self) -> None:
        self._context: dict[str, Any] = {}

    def set_context(self, ctx: dict[str, Any]) -> None:
        """Set context keys (replaces existing context)."""
        filtered = {k: v for k, v in ctx.items() if k != "auth.id"}
        self._context = filtered

    def get_context(self) -> dict[str, Any]:
        """Return a copy of the current context."""
        return dict(self._context)

    def clear_context(self) -> None:
        """Clear all context keys."""
        self._context = {}
